extensions/scarb-doc/src/docs_generation/markdown.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::{Context, Result};
+use camino::Utf8Path;
+use std::fs;
+
+use crate::docs_generation::markdown::all_items_file::generate_all_items_file_content;
+use crate::docs_generation::markdown::book_toml::generate_book_toml_content;
+use crate::docs_generation::markdown::summary::generate_summary_file_content;
+use crate::docs_generation::markdown::traits::{MarkdownDocItem, TopLevelMarkdownDocItem};
+use crate::docs_generation::{collect_all_top_level_items, TopLevelItems};
+use crate::types::Crate;
+
+mod all_items_file;
+mod book_toml;
+mod summary;
+mod traits;
+
+const BASE_HEADER_LEVEL: usize = 1;
+const SOURCE_DIRECTORY: &str = "src";
+const BOOK_TOML_FILENAME: &str = "book.toml";
+const SUMMARY_FILENAME: &str = "SUMMARY.md";
+const ALL_ITEMS_FILENAME: &str = "all_items.md";
+
+pub fn generate_and_save_markdown_docs(crate_: &Crate, output_dir: &Utf8Path) -> Result<()> {
```

Comment:
It's good practice to decouple business logic (_generation_) from IO (_saving_)

Reply:
Done